packages/editor/src/store/selectors.js

```diff
@@ -1796,3 +1796,34 @@ export const getPostTypeLabel = createRegistrySelector(
 export function isPublishSidebarOpened( state ) {
 	return state.publishSidebarActive;
 }
+
+/**
+ * Similar to getBlocksByName in @wordpress/block-editor, but only returns the top-most
+ * blocks that aren't descendants of the query block.
+ *
+ * @param {Object}       state      Global application state.
+ * @param {Array|string} blockNames Block names of the blocks to retrieve.
+ *
+ * @return {Array} Block client IDs.
+ */
+export const getPostBlocksByName = createRegistrySelector(
+	( select ) => ( state, blockNames ) => {
```

Comment:
I agree that making this a private selector makes sense for now.

We should also memoize the selector, but `blockNames` being an array might make that more challenging. Example: https://github.com/WordPress/gutenberg/pull/63732/files#diff-cf33f4589314d4a81e5a821d1433b83521dca9ad19d58a4b5ba1687e314b9558R35-R37

Reply:
Hmm yeah this is tricky to get right. Please double check f2fea66bebe for me 😅

Reply:
I wonder if making a selector variadic function would work here.

Something like:
```
export const getPostBlocksByName = ( state, ...blockNames );
getPostBlocksByName( ...contentOnlyBlockTypes );
```

Reply:
I don't want to stray far from `getBlocksByName` though.

Reply:
It's not a block for this PR. It's just an idea; I'm happy to give it a try separately.

Memoizing similar selector is always a plain 😅